### PR TITLE
Add redis-client support

### PR DESCRIPTION
**Pull request by jbockler**
With version 5 of the redis gem the client was moved into a separate gem. Since then the redis integration stopped working.

### CODE REVIEW

File: .semaphore/semaphore.yml [hunk 1]
```diff
@@ -500,6 +500,42 @@ blocks:
...
+        value: redis-4
+      - name: BUNDLE_GEMFILE
+        value: gemfiles/redis-4.gemfile
+      - name: _RUBYGEMS_VERSION
+        value: latest
+      - name: _BUNDLER_VERSION
+        value: latest
+      commands:
+      - "./support/bundler_wrapper exec rake test"
+    - name: Ruby 2.7.8 for redis-5
```
Comment (tombruijn) [inline .semaphore/semaphore.yml:521]: The build failure for this job seems quite consistent: https://appsignal.semaphoreci.com/jobs/d5fa2890-3c0c-4a57-9ef6-8eb7f1992292

Is there a known issue with hiredis on Ruby 2.7?

Comment (jbockler) [reply .semaphore/semaphore.yml:521]: Unfortunately, I haven't found anything about this, and I can't reproduce it either. The entire build matrix works perfectly on my local machine. So I'm a bit stuck here.

Comment (unflxw) [reply .semaphore/semaphore.yml:521]: Did some research on this. Long story short, there's something wrong with the OpenSSL extension in the build of Ruby 2.7.8 that Semaphore installs from its CDN. Manually re-building Ruby on the Semaphore instance directly (`rbenv install 2.7.8`) fixes the issue. A potential fix would be to do that and cache the resulting Ruby build across Semaphore job executions.

That said, the issue does not take place if we [upgrade the Semaphore machine to use Ubuntu 20.04](https://appsignal.semaphoreci.com/workflows/725495fb-3db4-4c36-a42d-753f8c2acfff). Instead, a completely different issue on a different library occurs (🙃) but if we were to sort that one out, then at least we would have upgraded the CI to a newer Ubuntu version.

**Review by tombruijn — APPROVED** [review]
We should also make a [test setup](https://github.com/appsignal/test-setups) for this so we can see it in action.

**Review by jeffkreeftmeijer — APPROVED** [review]

**Review by unflxw — APPROVED** [review]
Thanks for your contribution @jbockler!

I noticed that none of our Ruby [test setups](https://github.com/appsignal/test-setups/) used Redis directly, so I have [added Redis to a Ruby test setup](https://github.com/appsignal/test-setups/pull/183), to more easily test this integration now and in the future.

EDIT: 🤦 I just noticed that Tom mentioned it already 😅

**Pull request merged**